src/signer/mod.rs

```diff
@@ -50,8 +148,78 @@ impl AccountKeyPair {
     }
 }
 
+/// A trait for implementing custom signing logic.
+///
+/// This trait provides the core functionality needed to sign transactions and delegate actions.
+/// It is used by the [`Signer`] to abstract over different signing methods (secret key, ledger, keystore, etc.).
+///
+/// # Examples
+///
+/// ## Implementing a custom signer
+/// ```rust,no_run
+/// use near_api::{signer::*, types::{transactions::PrepopulateTransaction, CryptoHash}, errors::SignerError};
+/// use near_crypto::{PublicKey, SecretKey};
+/// use near_primitives::transaction::Transaction;
+///
+/// struct CustomSigner {
+///     secret_key: SecretKey,
+/// }
+///
+/// #[async_trait::async_trait]
+/// impl SignerTrait for CustomSigner {
+///     fn tx_and_secret(
+///         &self,
+///         tr: PrepopulateTransaction,
+///         public_key: PublicKey,
+///         nonce: u64,
+///         block_hash: CryptoHash,
+///     ) -> Result<(Transaction, SecretKey), SignerError> {
+///         let mut transaction = Transaction::new_v0(
+///             tr.signer_id.clone(),
+///             public_key,
+///             tr.receiver_id,
+///             nonce,
+///             block_hash.into(),
+///         );
+///         *transaction.actions_mut() = tr.actions;
+///         Ok((transaction, self.secret_key.clone()))
+///     }
+///
+///     fn get_public_key(&self) -> Result<PublicKey, SignerError> {
+///         Ok(self.secret_key.public_key())
+///     }
+/// }
+/// ```
+///
+/// ## Using a custom signer
+/// ```rust,no_run
+/// # use near_api::{signer::*, types::{transactions::PrepopulateTransaction, CryptoHash}, errors::SignerError};
+/// # use near_crypto::{PublicKey, SecretKey};
+/// # struct CustomSigner;
+/// # impl CustomSigner {
+/// #     fn new(_: SecretKey) -> Self { Self }
+/// # }
+/// # #[async_trait::async_trait]
+/// # impl SignerTrait for CustomSigner {
+/// #     fn tx_and_secret(&self, _: PrepopulateTransaction, _: PublicKey, _: u64, _: CryptoHash,
+/// #     ) -> Result<(near_primitives::transaction::Transaction, SecretKey), SignerError> { unimplemented!() }
+/// #     fn get_public_key(&self) -> Result<PublicKey, SignerError> { unimplemented!() }
+/// # }
+/// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
+/// let secret_key = "ed25519:...".parse()?;
+/// let custom_signer = CustomSigner::new(secret_key);
+/// let signer = Signer::new(custom_signer)?;
+/// # Ok(())
+/// # }
+/// ```
 #[async_trait::async_trait]
 pub trait SignerTrait {
+    /// Signs a delegate action for meta transactions.
+    ///
+    /// This method is used for meta-transactions where one account can delegate transaction delivery and gas payment to another account.
+    /// The delegate action is signed with a maximum block height to ensure the delegation expiration after some point in time.
+    ///
+    /// The default implementation should work for most cases.
     async fn sign_meta(
```

Comment:
Would be great to add an example where you have to implement provided methods

Reply:
Ok, maybe add something like "The example for reimplementation of provided methods, you can see ledger provider"